src/main/java/ca/on/oicr/gsi/dimsum/service/CaseService.java

```diff
@@ -111,6 +110,30 @@ public TableData<Case> getCases(int pageSize, int pageNumber, CaseSort sort, boo
     return data;
   }
 
+  public Set<String> getMatchingAssayNames(String prefix) {
+    return caseData.getAssayNames().stream()
+        .filter(s -> s.toLowerCase().startsWith(prefix.toLowerCase()))
+        .collect(Collectors.toSet());
+  }
+
+  public Set<String> getMatchingRequsitionNames(String prefix) {
+    return caseData.getRequisitionNames().stream()
+        .filter(s -> s.toLowerCase().startsWith(prefix.toLowerCase()))
+        .collect(Collectors.toSet());
+  }
+
+  public Set<String> getMatchingProjectNames(String prefix) {
+    return caseData.getProjectNames().stream()
+        .filter(s -> s.toLowerCase().startsWith(prefix.toLowerCase()))
+        .collect(Collectors.toSet());
+  }
+
+  public Set<String> getMatchingDonorNames(String prefix) {
+    return caseData.getDonorNames().stream()
+        .filter(s -> s.toLowerCase().startsWith(prefix.toLowerCase()))
+        .collect(Collectors.toSet());
+  }
```

Comment:
Add a private `getMatchingNames(Set<String>, String)` method to reduce the repetition here